File: malevich-web/src/app/_transfer/artworkDto.ts
import {CategoryDto} from "./categoryDto";
import {ArtistDto} from "./artistDto";
import {FileDto} from "../../../node_modules/yinyang-core";

export class ArtworkDto {
  id: number;
  estimatedPrice: number;
  category: CategoryDto;
  artist: ArtistDto;
  thumbnail: FileDto;
  image: FileDto;
  titleMl: Map<string, string>;
  descriptionMl: Map<string, string>;
}

File: malevich-server/src/main/java/io/malevich/server/transfer/ArtworkDto.java
package io.malevich.server.transfer;


import com.yinyang.core.server.transfer.FileDto;
import lombok.Getter;
import lombok.Setter;

import javax.xml.bind.annotation.XmlRootElement;
import java.util.Map;

@Getter
@Setter
@XmlRootElement(name = "artwork")
public class ArtworkDto {

    private Long id;

    private Double estimatedPrice;

    private CategoryDto category;

    private ArtistDto artist;

    private FileDto thumbnail;

    private FileDto image;

    private Map<String, String> titleMl;

    private Map<String, String> descriptionMl;

}
